File: src/main/java/xyz/avarel/waffles/nodes/basic/IntNode.java
package xyz.avarel.waffles.nodes.basic;

import com.oracle.truffle.api.frame.VirtualFrame;
import xyz.avarel.waffles.nodes.WaffleNode;

public class IntNode extends WaffleNode {
    private final int value;

    public IntNode(int value) {
        this.value = value;
    }

    @Override
    public Object execute(VirtualFrame virtualFrame) {
        return value;
    }

    @Override
    public int executeInt(VirtualFrame virtualFrame) {
        return value;
    }

    @Override
    public String toString() {
        return Integer.toString(value);
    }
}

File: src/main/kotlin/xyz/avarel/waffles/nodes/basic/IntNode.kt
package xyz.avarel.waffles.nodes.basic

import com.oracle.truffle.api.frame.VirtualFrame
import xyz.avarel.waffles.nodes.WaffleNode

class IntNode(private val value: Int) : WaffleNode() {
    override fun execute(virtualFrame: VirtualFrame) = value

    override fun executeInt(virtualFrame: VirtualFrame) = value

    override fun toString() = value.toString()
}
